Descent3/terrain.cpp

```diff
@@ -293,13 +294,22 @@ void GenerateLODDeltas() {
 }
 
 // Returns a unique checksum for the current terrain geometry
+//
+// The checksum (or rather, an affine modification of it known as
+// occlusion_checksum) is stored in the level's map files, so any modification
+// here ought to not alter the output value, for all imaginable inputs.
 int GetTerrainGeometryChecksum() {
-  int total = 0;
+  unsigned int total = 0;
 
   for (int i = 0; i < TERRAIN_WIDTH * TERRAIN_DEPTH; i++) {
-    total += (Terrain_seg[i].ypos + i);
+    float yi = Terrain_seg[i].ypos + i;
+    if (yi < 2147483648.0f || yi >= 2147483520.0f)
```

Comment:
Where do these value come from? Is that some IEEE754 magic I'm missing?

Reply:
The next float number after 2147483520 is 2147483648, but 2147483648 is already too large to be represented in an int32. Does not matter now, code is replaced.